src/all/pixiv/src/eu/kanade/tachiyomi/extension/all/pixiv/Common.kt

```diff
@@ -0,0 +1,43 @@
+package eu.kanade.tachiyomi.extension.all.pixiv
+
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.model.UpdateStrategy
+import kotlinx.serialization.json.Json
+import org.jsoup.Jsoup
+import uy.kohesive.injekt.injectLazy
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+internal const val BASE_URL = "https://www.pixiv.net"
+internal val json: Json by injectLazy()
+
+private val dateFormat by lazy { SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss", Locale.JAPAN) }
+private val illustIdRegex by lazy { Regex("/artworks/(\\d+)") }
+
+internal fun getIllustIdFromUrl(url: String): Int =
+    illustIdRegex.find(url)?.groupValues?.get(1)?.toInt()
+        ?: throw IllegalArgumentException("$url is not a valid Pixiv URL")
+
+internal fun parseTimestamp(string: String) =
+    runCatching { dateFormat.parse(string)?.time!! }.getOrDefault(0)
+
+internal fun PixivArtwork.toSManga() = SManga.create().apply {
+    val artwork = this@toSManga
+
+    url = "$BASE_URL/artworks/${artwork.id}"
```

Comment:
Should not include the baseURL in the `url` fields to prevent future issues if ever the source changes their URL.

```suggestion
    url = "/artworks/${artwork.id}"
```